### ISSUE TITLE
Getting the example to work 

### ISSUE THREAD

**Issue by Megatronicus** [tensorflow/flutter-tflite#119]
Hi, I've tried using this example 

https://manojbhor.medium.com/nsfw-classifier-nudity-classification-on-mobile-and-edge-devices-bae14c171a89

I've put the tflite file and labels.txt into your official flutter_tflite github example, however it's not classifying any image right and sometimes I get infinity. 

Also, for the example to work I had to change this 

  // final output = [List<int>.filled(isolateModel.outputShape[1], 0)]; ///ORIGINAL CODE
      final output = [List<double>.filled(isolateModel.outputShape[1], 0)];
      and this 
            // int maxScore = result.reduce((a, b) => a + b);///ORIGINAL CODE
      int maxScore = result.reduce((a, b) => a + b).toInt();
      
      Othewise I would get an error 
      
      D/EGL_emulation(29366): app_time_stats: avg=2443.00ms min=92.17ms max=6949.70ms count=3
I/tflite  (29366): Replacing 71 node(s) with delegate (TfLiteXNNPackDelegate) node, yielding 1 partitions for the whole graph.
[log] Interpreter loaded successfully
W/ation_mobilenet(29366): Reducing the number of considered missed Gc histogram windows from 185 to 100
E/flutter (29366): [ERROR:flutter/runtime/dart_isolate.cc(1097)] Unhandled exception:
E/flutter (29366): type 'List<double>' is not a subtype of type 'List<int>' of 'value'
E/flutter (29366): #0      List.[]= (dart:core-patch/growable_array.dart)
E/flutter (29366): #1      Tensor._duplicateList (package:tflite_flutter/src/tensor.dart:237:10)
E/flutter (29366): #2      Tensor.copyTo (package:tflite_flutter/src/tensor.dart:203:7)
E/flutter (29366): #3      Interpreter.runForMultipleInputs (package:tflite_flutter/src/interpreter.dart:183:24)
E/flutter (29366): #4      Interpreter.run (package:tflite_flutter/src/interpreter.dart:172:5)
E/flutter (29366): #5      IsolateInference.entryPoint (package:image_classification_mobilenet/helper/isolate_inference.dart:86:19)
E/flutter (29366): <asynchronous suspension>

However, maybe exactly due to this change, or possibly something else I'm getting false readings for example for an image of a room it says it's a drawing and then some nsfw images it labels as drawings, basically the first word in the labels.txt is the one that's being chosen most of the time, either that or I get infinity. 

Can you help me put that example together so that it works correctly instead of getting false readings? 

There's also an interesting comment left for the article, the user had the same problem as I'm experiencing right now, maybe it helps

Hi Harshit Dwivedi,
Thanks a lot for the article. I would be grateful if you could have time to answer following question.
I am getting some wrong output values from my code if I use your model.tflite file.It does not matter, which photo is classified, somehow the first item(drawings) gets always the highest value.
EX: [[9.9424970e-01 5.7404796e-03 1.7507012e-06 8.0635009e-06]]
Do you have an idea what the reason could be.?
I followed the steps below for the object classification:
https://heartbeat.fritz.ai/running-tensorflow-lite-image-classification-models-in-python-92ef44b4cd47
the only change which I had to do because of runtime errors(I guess because of tftite file differences):
Instead of following line:
interpreter.set_tensor(input_details[0]['index'], [new_img])
Used these lines:
input_shape = input_details[0]['shape’]input_data1 = np.array(new_img, dtype=np.float32)input_data1 = np.reshape(input_data1,input_shape)interpreter.set_tensor(input_details[0]['index'], input_data1)
My Environment:
MacOS
Tensorflow 1.13.1
input_details of model.tflite file:
[{'name': 'mobilenetv2_1_00_224_input', 'index': 0, 'shape': array([ 1, 224, 224, 3], dtype=int32), 'dtype': <class 'numpy.float32'>, 'quantization': (0.0, 0)}]
output_details of model.tflite file:
[{'name': 'Identity', 'index': 181, 'shape': array([1, 4], dtype=int32), 'dtype': <class 'numpy.float32'>, 'quantization': (0.0, 0)}]
Note: 1-I found another tflite file in internet, which has following input and output values(which is similar with the values defined in the link above) and works also totally fine:https://github.com/joonb14/TFLiteClassification
[{'name': 'input', 'index': 88, 'shape': array([ 1, 224, 224, 3], dtype=int32), 'dtype': <class 'numpy.uint8'>, 'quantization': (0.0078125, 128)}]
[{'name': 'MobilenetV1/Predictions/Reshape_1', 'index': 87, 'shape': array([ 1, 1001], dtype=int32), 'dtype': <class 'numpy.uint8'>, 'quantization': (0.00390625, 0)}]
Note 2-If you need my source code, I can provide you per email.
Thank you very much in advance

Thank you.

**Comment by Megatronicus** [comment]
Can you guys please help me?

**Comment by Megatronicus** [comment]
I tried it on a phone instead of an emulator, then I didn't have to change this code 

// final output = [List.filled(isolateModel.outputShape[1], 0)]; ///ORIGINAL CODE
final output = [List.filled(isolateModel.outputShape[1], 0)];
and this
// int maxScore = result.reduce((a, b) => a + b);///ORIGINAL CODE
int maxScore = result.reduce((a, b) => a + b).toInt();

However it's still giving out wrong result that each image is the first option in the list of labels, either that or it says infinity for all optins in labels text.

**Comment by PaulTR** [comment]
Yeah definitely use a real device over an emulator. That person's tutorial may also be out of date since there was a large refactor recently. Did you train the model that's being used yourself, or is it one you grabbed from that article 2 years ago? I'm wondering if it has to do with a trained-for version of TFLite and the current version that we're supporting. You can also try it directly with an existing Android sample (something like [this](https://github.com/tensorflow/examples/tree/master/lite/examples/image_classification/android))

**Comment by Megatronicus** [comment]
Can you provide a codelab or a tutorial where I can train the model, I didn't train these myself, I found them on the web like that provided link above and some I converted from jpynb keras using python to convert to tflite but that didn't work either. I might have to then train my own model since like you say these might all be old models that don't work anymore after newest changes. Unfortunately, I don't need a specific android sample, I need a flutter sample to use with tflite_flutter, one that supports android, ios and flutter web. Can you help me please?

**Comment by PaulTR** [comment]
The official docs for model training will be a good starting place - that's really outside of the scope of this repo (and my own knowledge).

The examples folder in this repo has all of the tflite plugin samples available that you can use to build your feature, though web isn't a part of the support.

**Comment by Megatronicus** [comment]
"The official docs for model training will be a good starting place"
Can you please post a link to the official docs? 

"The examples folder in this repo has all of the tflite plugin samples available that you can use to build your feature, though web isn't a part of the support."

Ok thank you for responding! 
Best regards

**Comment by PaulTR** [comment]
Your best bet is doing a search for "TensorFlow model maker" :) You'll get [something like this](https://www.tensorflow.org/lite/models/modify/model_maker) as the first result.

**Comment by Megatronicus** [comment]
Found out I had that in my bookmarks already. Ok, will try their tutorials, thanks! =)
